src/test/scala/de/upb/cs/uc4/hyperledger/tests/UnsignedTransactionTests.scala

```diff
@@ -33,42 +37,76 @@ class UnsignedTransactionTests extends TestBase {
         val proposal: Proposal = Proposal.parseFrom(proposalBytes)
         println("\n\n\n##########################\nHeader:\n##########################\n\n" + proposal.getHeader.toStringUtf8)
         println("\n\n\n##########################\nPayload:\n##########################\n\n" + proposal.getPayload.toStringUtf8)
       }
     }
 
     "passing a signed transaction" should {
+      "submit the proposal transaction to the proposal contract, even if the signature was not created using the private key belonging to the connection" in {
+        val argEnrollmentId = "frontend-signing-tester"
+        val testAffiliation = organisationId
+        super.tryEnrollment(caURL, tlsCert, walletPath, username, password, organisationId, channel, chaincode, networkDescriptionPath)
+        // initialize crypto primitives
+        val crypto: CryptoPrimitives = TestHelper.getCryptoPrimitives()
+        // get testUser certificate and private key
+        val testUserIdentity: X509IdentityImpl = tryRegisterAndEnrollTestUser(argEnrollmentId, testAffiliation)
+        val privateKey: PrivateKey = testUserIdentity.getPrivateKey
+        val certificate: X509Certificate = testUserIdentity.getCertificate
+        val argCertificate = TestHelper.toPemString(certificate)
+        println("\n\n\n##########################\nCertificate:\n##########################\n\n" + argCertificate)
+        // get proposal
+        val proposalBytes = certificateConnection.getProposalAddCertificate(Identities.toPemString(certificate), testAffiliation, argEnrollmentId, argCertificate)
+        println("\n\n\n##########################\nProposal Bytes:\n##########################\n\n" + Base64.getEncoder.encodeToString(proposalBytes))
+        // sign proposal with testUser privateKey
+        val signatureBytes = crypto.sign(privateKey, proposalBytes)
+        val b64Sig = ByteString.copyFrom(Base64.getEncoder.encode(signatureBytes)).toStringUtf8
+        println("\n\n\n##########################\nSignature:\n##########################\n\n" + b64Sig)
+        val transactionPayloadBytes: Array[Byte] = certificateConnection.getUnsignedTransaction(proposalBytes, signatureBytes)
+        println("\n\n\n##########################\nTransactionPayload:\n##########################\n\n" + Base64.getEncoder.encodeToString(transactionPayloadBytes))
+        val transactionSignature: Array[Byte] = crypto.sign(privateKey, transactionPayloadBytes)
+        val result = certificateConnection.submitSignedTransaction(transactionPayloadBytes, transactionSignature)
+        println("\n\n\n##########################\nResult:\n##########################\n\n" + result)
+        val getResult = initializeApproval(username).getApprovals("UC4.Certificate", "addCertificate", argEnrollmentId, argCertificate)
+        println("\n\n\n##########################\nCompareResult:\n##########################\n\n" + getResult)
```

Comment:
Is this amount of `println` in test actually necessary?

Reply:
A `yes` is a valid answer here btw

Reply:
currently resolving / reworking the tests
instead of printing these and checking manually during the bugfixing/development we need to automatically check for expected contents

Reply:
https://github.com/upb-uc4/hlf-api/pull/126